### PR TITLE
Feature/payment method already paid

### PR DESCRIPTION
**Pull request by Dominik2809**
### All changes meet the following requirements
- [x] Changelog entry was added
- [ ] Changes have been documented
- [x] Changes have been tested by the author
- [x] Changes have been tested by the reviewer

@plentymarkets/plentyshop

### CODE REVIEW

File: src/Config/CeresCheckoutConfig.php [hunk 1]
```diff
@@ -21,6 +21,12 @@ class CeresCheckoutConfig extends PluginConfig
      */
     public $showAllShippingProfiles;
 
+
+    /**
+     * @var string $alreadyPaidIconUrl
+     */
+    public $alreadyPaidIconUrl;
```
Comment (stentrop) [inline src/Config/CeresCheckoutConfig.php:28]: if we can we should use types

File: src/Methods/AlreadyPaidPaymentMethod.php [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+use Plenty\Modules\Frontend\Contracts\Checkout;
+use Plenty\Modules\Payment\Method\Services\PaymentMethodBaseService;
+use Plenty\Plugin\Application;
+use Plenty\Plugin\Translation\Translator;
+use Ceres\Config\CeresConfig;
+
+class AlreadyPaidPaymentMethod extends PaymentMethodBaseService
+{
+    /** @var BasketRepositoryContract */
+    private $basketRepository;
```
Comment (stentrop) [inline src/Methods/AlreadyPaidPaymentMethod.php:17]: types ?

File: src/Wizard/ShopWizard/Repositories/AlreadyPaidShippingCountryRepository.php [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+        return false;
+    }
+
+    /**
+     * @param $plentyId
+     * @param $shippingCountryIds
+     * @return bool
+     */
+    public function saveShippingCountries(int $plentyId, array $shippingCountryIds)
```
Comment (stentrop) [inline src/Wizard/ShopWizard/Repositories/AlreadyPaidShippingCountryRepository.php]: return types

File: src/Wizard/ShopWizard/Services/AlreadyPaidShippingCountriesService.php [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+class AlreadyPaidShippingCountriesService
+{
+
+    /**
+     * @param int $plentyId
+     * @param array $shippingCountriesIds
+     * @return void
+     */
+    public function execute(int $plentyId, array $shippingCountriesIds)
```
Comment (stentrop) [inline src/Wizard/ShopWizard/Services/AlreadyPaidShippingCountriesService.php:15]: so we always delete all entries ?

File: src/Wizard/ShopWizard/Services/SettingsHandlerService.php [hunk 5]
```diff
@@ -22,4 +23,16 @@ public function getStoreIdentifier($webstoreId)
 
         return $storeIdentifier;
     }
+
+    /**
+     * @param $plentyId
+     * @return array
+     */
+    public function getAlreadyPaidShippingCountries($plentyId)
```
Comment (stentrop) [inline src/Wizard/ShopWizard/Services/SettingsHandlerService.php]: type

File: src/Wizard/ShopWizard/Steps/Builder/OnlineStoreStep.php [hunk 6]
```diff
@@ -512,4 +551,38 @@ private function buildLoginModeSettings()
...
+     * @return array
+     */
+    private function getCountriesListForm()
+    {
+        if ($this->deliveryCountries === null) {
+            /** @var CountryRepositoryContract $countryRepository */
+            $countryRepository = pluginApp(CountryRepositoryContract::class);
+            $countries = $countryRepository->getCountriesList(true, ['names']);
+            $this->deliveryCountries = [];
+            $systemLanguage = $this->getLanguage();
```
Comment (stentrop) [inline src/Wizard/ShopWizard/Steps/Builder/OnlineStoreStep.php]: systemLanguage !==  LocaleLanguage

**Review by stentrop — CHANGES_REQUESTED** [review]

File: src/Wizard/ShopWizard/Services/AlreadyPaidShippingCountriesService.php (hunk 4 above)
Comment (Dominik2809) [reply src/Wizard/ShopWizard/Services/AlreadyPaidShippingCountriesService.php:15]: yes

**Review by Tim-M-S — APPROVED** [review]

**Pull request merged**